test/support/src/vfs_helpers.cc

```diff
@@ -415,5 +442,132 @@ std::string TemporaryDirectoryFixture::create_temporary_array(
   return array_uri;
 }
 
+VFSTest::VFSTest(
+    const std::vector<size_t>& test_tree, const std::string& prefix)
+    : test_tree_(test_tree)
+    , compute_(4)
+    , io_(4)
+    , vfs_(
+          &tiledb::test::g_helper_stats,
+          &io_,
+          &compute_,
+          tiledb::test::create_test_config())
+    , temp_dir_(tiledb::test::test_dir(prefix)) {
```

Comment:
I didn't have a use for `ensure_supported()` at the moment since the only IO functions are part of test setup which goes back to my other comment on not being able to construct the test classes without failing TEMPLATE_LIST_TEST_CASE from throwing in functions called by the constructors. I can still add it if we want to leave it unused and document it's purpose.

The is_supported member and rename is done though, and I added a note to throw in the `VFSTestBase` constructor when we can produce a list of supported filesystems and pass to Catch2.